samples/FormatFilterSample.Web/Controllers/FormatFilterController.cs

```diff
@@ -21,5 +22,21 @@ public Product ProducesMethod(int id)
         {
             return new Product() { SampleInt = id };
         }
+
+        [Route("[controller]/[action]/{id}.{format}")]
+        public IActionResult ExplicitContentTypeOnObjectResult(int id)
+        {
+            var result = new ObjectResult(new Product() { SampleInt = id });
+            result.ContentTypes.Add(new MediaTypeHeaderValue("text/xml"));
+            return result;
+        }
+
+        [Route("[controller]/[action]/{id}.{format}")]
+        public Product ExplicitContentTypeOnResponse(int id)
+        {
+            Response.ContentType = "text/xml";
+
+            return new Product() { SampleInt = id };
```

Comment:
Good point. This should not be in a sample. This scenario is covered in unit test here so we should be good to remove it.